src/components/axes/axis.ts

```diff
@@ -188,7 +188,7 @@ module Plottable {
     }
 
     protected _generateTickMarkAttrHash(isEndTickMark = false) {
-      var tickMarkAttrHash = {
+      var tickMarkAttrHash: { [key: string]: number | ((d: any) => number) } = {
```

Comment:
Let's keep the union, but isn't the second argument `[key: string]: (d:any) => number`? Like, it's a map from key to functions? I'm missing something here. How does this even compile?